internal/bminventory/inventory_test.go

```diff
@@ -13048,3 +13048,90 @@ func validateNetworkConfiguration(cluster *models.Cluster, clusterNetworks *[]*m
 		}
 	}
 }
+
+var _ = Describe("Update cluster - feature usage flags", func() {
+	var (
+		bm      *bareMetalInventory
+		cfg     Config
+		db      *gorm.DB
+		cluster *common.Cluster
+		dbName  string
+		usages  = map[string]models.Usage{}
+	)
+	BeforeEach(func() {
+		Expect(envconfig.Process("test", &cfg)).ShouldNot(HaveOccurred())
+		Expect(cfg.IPv6Support).Should(BeTrue())
+		cfg = Config{}
+		db, dbName = common.PrepareTestDB()
+		bm = createInventory(db, cfg)
+		cluster = createCluster(db, models.ClusterStatusPendingForInput)
+	})
+
+	AfterEach(func() {
+		common.DeleteTestDB(db, dbName)
+	})
+
+	Context("CPU Architecture usage", func() {
+		It("Add feature usage of ARM64 when cluster arch match", func() {
+			///
+			mockUsage.EXPECT().Add(usages, usage.CPUArchitectureARM64, nil).Times(1)
+			mockUsage.EXPECT().Remove(usages, usage.CPUArchitectureARM64).Times(0)
+			cluster.CPUArchitecture = "arm64"
+			bm.updateClusterCPUFeatureUsage(cluster, usages)
+		})
+
+		It("Remove feature usage of ARM64 when cluster arch match", func() {
+			///
+			mockUsage.EXPECT().Add(usages, usage.CPUArchitectureARM64, nil).Times(0)
+			mockUsage.EXPECT().Remove(usages, usage.CPUArchitectureARM64).Times(1)
+			bm.updateClusterCPUFeatureUsage(cluster, usages)
+		})
+	})
+
+	Context("Cluster managed network with VMs", func() {
+		addVM := func(cluster *common.Cluster) {
+			hostID := strfmt.UUID(uuid.New().String())
+			infraEnv := createInfraEnv(db, strfmt.UUID(uuid.New().String()), *cluster.ID)
+			inventory := models.Inventory{
+				SystemVendor: &models.SystemVendor{
+					Virtual: true,
+				},
+			}
+			inventoryByte, err := json.Marshal(inventory)
+			Expect(err).ToNot(HaveOccurred())
+			host := addHost(hostID, models.HostRoleAutoAssign, models.HostStatusKnown, models.HostKindHost,
+				infraEnv.ID, *cluster.ID, string(inventoryByte), db)
+			cluster.Hosts = append(cluster.Hosts, &host)
+		}
```

Comment:
any reason not to put it top-level with ``func addVM(...``
it's also easier adding a short doc-string